### ISSUE TITLE
NPM Scripts bar does not appear in sidebar explorer

### ISSUE THREAD

**Issue by leosdad** [microsoft/vscode#104594]
Issue Type: <b>Bug</b>

The NPM Scripts bar does not appear anymore for some of my workspaces, while it appears in others for no apparent reason. I have scripts in all projects, they are in the "scripts" section of package.json.

It seems this this problem appeared in this latest update.

VS Code version: Code 1.48.0 (db40434f562994116e5b21c24015a2e40b2504e6, 2020-08-13T07:50:42.600Z)
OS version: Windows_NT x64 10.0.18363

<details><summary>Extensions (26)</summary>

Extension|Author (truncated)|Version
---|---|---
overtype|ada|0.2.0
vscode-css-formatter|aes|1.0.1
path-intellisense|chr|2.2.1
jsrefactor|cms|2.20.6
file-tree-to-text-generator|d-k|1.2.1
vscode-eslint|dba|2.1.8
LogFileHighlighter|emi|2.9.0
nunjucks-template|ese|0.3.1
git-commits|Exe|3.7.4
vscode-highlight|fab|1.4.1
todo-tree|Gru|0.0.178
git-last-commit-message|Jan|1.9.0
vue|jcb|0.1.5
hjson|lak|0.3.2
vetur|oct|0.26.1
vscode-sort-json|ric|1.18.0
live-sass|rit|3.0.0
LiveServer|rit|5.6.1
nunjucks|ron|0.3.0
vscode-scss-formatter|sib|2.0.1
code-spell-checker|str|1.9.0
code-spell-checker-portuguese-brazilian|str|0.1.2
language-stylus|sys|1.11.0
es6-string-html|Tob|1.8.8
txt-syntax|xsh|0.0.4
markdown-all-in-one|yzh|3.2.0

</details>

**Comment by RudiVuuren** [comment]
@leosdad, just a friendly note: The **NPM Scripts** section won't appear if you don't have a **package.json** in the root directory of your workspace.

**Comment by leosdad** [comment]
> @leosdad, just a friendly note: The **NPM Scripts** section won't appear if you don't have a **package.json** in the root directory of your workspace.

Thanks. As said above, I have **package.json** in all projects.

**Comment by gjsjohnmurray** [comment]
@leosdad are you able to get the NPM Scripts view to appear by clicking the "..." on the EXPLORER header bar and picking it from the menu?

I think some work was done in 1.48 affecting whether this view shows by default.

**Comment by gjsjohnmurray** [comment]
See also #104591

**Comment by leosdad** [comment]
> @leosdad are you able to get the NPM Scripts view to appear by clicking the "..." on the EXPLORER header bar and picking it from the menu?
> 
> I think some work was done in 1.48 affecting whether this view shows by default.

Thanks, but no. View / Open View also does not show it.

I'll break it all down using four different projects of mine. Here's a sample from project 1 ("web"):

![image](https://user-images.githubusercontent.com/7026091/90249344-ad815a00-de10-11ea-8801-267fac830a20.png)

![image](https://user-images.githubusercontent.com/7026091/90248217-a78a7980-de0e-11ea-83b1-d0c30c8b6cbb.png)

Here's a snippet of its package.json:

```
{
  "scripts": {
    "serve": "vue-cli-service serve --port 8085",
    "build": "vue-cli-service build",
    "lint": "vue-cli-service lint",
    "doc": "jsdoc --configure jsdoc.json"
  },
}
```
> I tried pasting all package.jsons, but this post became too long.

Project 2 ("ide") that has the same bug, i.e. it also does **not** show the NPM Scripts bar / menu.

On the other hand, here are screen captures from project # 3 ("builder") that has **no** NPM scripts:

![image](https://user-images.githubusercontent.com/7026091/90249008-1916f780-de10-11ea-82d0-a80557a08769.png)

![image](https://user-images.githubusercontent.com/7026091/90248160-89247e00-de0e-11ea-835d-f0090280b065.png)

Finally, here's the capture from project # 4 ("sat") that **does** have scripts:

![image](https://user-images.githubusercontent.com/7026091/90249490-f507e600-de10-11ea-8f0d-766edbd4997b.png)

To summarize:

| # | Project | Has scripts | Project.json | Shows NPM Scripts bar |
|----|----|----|----|----|
|1|web|yes|yes|**no**|
|2|ide|yes|yes|**no**|
|3|builder|no|yes|yes|
|4|sat|yes|yes|yes|

So that's not the same as issuea #104591. All projects use package.json and the issue must have another cause.

**Comment by gjsjohnmurray** [comment]
@alexr00 please reopen this and investigate.

**Comment by alexr00** [comment]
@leosdad thanks for the break down. Just to make sure I'm understanding the setup of "web" and "ide" correctly can you confirm: Do both "web" and "ide" have a package.json file _at the root of the project_? You clearly state that you have a package.json in all projects, but is it in a folder at the root? I'm sorry to be repetitive, but since I can't repro the issue I want to make sure I understand what's going on.

Are any of the projects you're seeing the issue on available publicly? I'd like to clone it and see if I can repro the issue.

**Comment by leosdad** [comment]
Hmmmm. Yes, there are differences. The ones where the Npm Scripts bar do not appear are the ones where package.json is not in the same folder as the .code-workspace file. But I don't recall this being a problem before, and I still think the bar should appear nonetheless. Is there any setting to inform the location of package.json?

**Comment by alexr00** [comment]
@leosdad its looking like you are seeing https://github.com/microsoft/vscode/issues/104591. The NPM scripts explorer should appear as shown below, but because of https://github.com/microsoft/vscode/issues/104591, it's missing from there. In workspaces where you previously had the NPM script explorer visible there should have been no change; however, https://github.com/microsoft/vscode/issues/104591 causes it to vanish from there too.

![recording (16)](https://user-images.githubusercontent.com/38270282/90230165-68adf100-de19-11ea-9ac8-e23173d962a0.gif)

The easiest way to determine if the fix for https://github.com/microsoft/vscode/issues/104591 will solve your issue is for you to install the[ VS Code Insiders](https://code.visualstudio.com/insiders/) build on Monday. It installs side by side with stable, so it's easy to try. Note that the NPM scripts view will only appear as in the gif above (instead of being visible in the sidebar) since you haven't used insiders before.

**Comment by leosdad** [comment]
Just did that, installed Insiders, but no dice. It's jist the same. In project "ide" all I get is:

![image](https://user-images.githubusercontent.com/7026091/90257249-4e761200-de1d-11ea-84e7-2b0ebf72dead.png)

And in project "web":

![image](https://user-images.githubusercontent.com/7026091/90257344-6f3e6780-de1d-11ea-92bc-403dd2244376.png)

It's okay with project "builder", though:

![image](https://user-images.githubusercontent.com/7026091/90257182-2f778000-de1d-11ea-99c8-3610d9e20a78.png)

**Comment by leosdad** [comment]
> Are any of the projects you're seeing the issue on available publicly? I'd like to clone it and see if I can repro the issue.

Unfortunately not. I can send you my project.json and .code-workspace files, though.

**Comment by alexr00** [comment]
@leosdad the fix was only merged today and isn't available in Insiders yet, which is why I suggested Monday.

**Comment by leosdad** [comment]
Oh, **next** Monday. Okay, I'll surely try it then. Many thanks for now.

**Comment by leosdad** [comment]
Yep, the insiders version works as it should. The NPM Scripts bar is working again.
